synapse/handlers/federation.py

```diff
@@ -683,6 +684,41 @@ async def _get_events_from_store_or_dest(
 
         return fetched_events
 
+    _forwarded_key = "net.maunium.msc2730.forwarded"
+
+    async def _validate_forwarded_event(
+        self, event: EventBase
+    ) -> Tuple[bool, Optional[str]]:
+        try:
+            source_evt_dict = {**event.content[self._forwarded_key]}
+            room_version_identifier = source_evt_dict["unsigned"]["room_version"]
+            source_evt_dict["type"] = event.type
+            source_evt_dict["content"] = {**event.content}
+            del source_evt_dict["unsigned"]
+            del source_evt_dict["content"][self._forwarded_key]
+        except (KeyError, TypeError):
+            return False, None
+
+        room_version = KNOWN_ROOM_VERSIONS.get(room_version_identifier)
+        if not room_version:
+            # We don't support the source room version, so we can't verify the event :(
+            return False, None
+
+        try:
+            source_evt = event_from_pdu_json(source_evt_dict, room_version)
+        except SynapseError:
+            return False, None
```

Comment:
as a general rule, eating exceptions like this without giving any clue about what the exception was leads to hard-to-debug failures. I'd recommend logging something before returning.